runtime/browser/runtime.cc

```diff
@@ -225,13 +222,10 @@ void Runtime::DidNavigateMainFramePostCommit(
 
 content::JavaScriptDialogManager* Runtime::GetJavaScriptDialogManager(
     content::WebContents* web_contents) {
-#if defined(OS_LINUX)
+#if defined(USE_AURA)
   return app_modal::JavaScriptDialogManager::GetInstance();
 #else
-  if (!javascript_dialog_manager_) {
-    javascript_dialog_manager_.reset(new RuntimeJavaScriptDialogManager);
-  }
-  return javascript_dialog_manager_.get();
```

Comment:
We can't remove RuntimeJavaScriptDialogManager as it's used in Android. Probably we may need to move Android's JS dialog implementation to based on app_modal.